File: banshee/src/models/data/form/ShinyFormSection.ts
import { ShinyFormField } from './ShinyFormField';

export class ShinyFormSection {
  title: string;
  fields: ShinyFormField[];
  collapsible: boolean;
  initiallyCollapsed: boolean;

  constructor(
    title: string,
    fields: ShinyFormField[],
    collapsible: boolean,
    initiallyCollapsed: boolean
  ) {
    this.title = title;
    this.fields = fields;
    this.collapsible = collapsible;
    this.initiallyCollapsed = initiallyCollapsed;
  }
}

File: src/main/java/io/vertigo/shiny/models/data/form/ShinyFormSection.java
package io.vertigo.shiny.models.data.form;

import java.util.List;

import io.vertigo.core.lang.Assertion;

public record ShinyFormSection(
		String title,
		List<ShinyFormField> fields,
		boolean collapsible,
		boolean initiallyCollapsed) {
	public ShinyFormSection {
		Assertion.check()
				.isNotBlank(title)
				.isNotNull(fields);
	}
}
